src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -804,6 +804,11 @@ public function testDumpNullAsTilde()
         $this->assertSame('{ foo: ~ }', $this->dumper->dump(['foo' => null], 0, 0, Yaml::DUMP_NULL_AS_TILDE));
     }
 
+    public function testDumpNumericKeyAsString()
+    {
+        $this->assertSame("{ '200': foo }", $this->dumper->dump([200 => 'foo'], 0, 0, Yaml::DUMP_NUMERIC_KEY_AS_STRING));
```

Comment:
You are only testing one happy case for this new feature. If we really want to introduce that flag, we will need a lot more tests.

Please cover the case @maxbeckers gave you in https://github.com/symfony/symfony/pull/48127/files#r1015074371. More cases that come to my mind:

* Dump a single integer, maybe also a string/float/bool/null. The flag must not affect the dumper's behavior for those cases.
* Dump a true like array (`['a', 'b', 'c']`). In #48061, @mabar came to the conclusion that we should not type-cast keys here.
* Dump an array with mixed-type keys (`[42 => 'a', 'b' => 'c']`).
* Dump a broken list (`['a', 'b', 42 => 'c']`)
* More edge cases  that come to your mind.

Reply:
Thanks for your detailed feedback, indeed my approach was naive. There are a lot of other cases (inline ou not, TaggedValues, ArrayObject).

I will try support them and ping reviewers when PR is "really" ready